jax/api.py

```diff
@@ -1966,6 +1966,40 @@ def device_put(x, device: Optional[xc.Device] = None):
   return tree_map(lambda y: xla.device_put_p.bind(y, device=device), x)
 
 
+def device_put_sharded(x, devices: Sequence[xc.Device]):
+  """Shards the input to specified devices, returning ShardedDeviceArrays.
+
+  Args:
+    x: A sequence of arrays, scalars, or (nested) standard Python containers thereof
```

Comment:
Not a sequence, right?

Reply:
I called this out as a sequence, because this function requires that the length of `x` be equal to the length of `devices`, to distinguish this from `device_put_replicated` in which there is no such constraint.

Reply:
Updated the description to be more clear.

Reply:
Ahh, ok, then I'm not sure that "Shards the input" is the right description, because it doesn't cut any input into pieces. Maybe "Creates a sharded device array representing the input stacked over devices", although I don't really like that language either?

Reply:
It does cut the input into pieces if the input is an array. If the input is a list, it's essentially treated as a list of pre-determined shards.

Reply:
Thanks, that clears things up for me. Allowing either an array or a list seems like a good approach here and one that's consistent with NumPy expectations, but it that should maybe be documented more explicitly in the JAX context since we don't do it as much.

Reply:
I'm happy to not allow an array here, because the functionality that downstream libraries need to turn a list of arrays into a stacked sharded array.

Reply:
Done